scania-digital-design-system/tegel
# feat(checkbox): indeterminate state

**Describe pull-request**  
This PR introduces a new prop for the `tds-checkbox` - `indeterminate` state.

The prop displays the checkbox as indeterminate, it has a higher prio than checked. If the checkbox is `indeterminate` that will take precedence over `checked`.

Since `indeterminate` is a state of the checkbox, we can't know when it should be indeterminate or not, therefor that will be in the hands of the users.

This PR also introduces basic unit and e2e tests for the checkbox.

**Solving issue**  
Fixes: [CDEP-1589](https://tegel.atlassian.net/browse/CDEP-1589)

**How to test**  
1. Go to Checkbox
2. Use the controls to set the checkbox to a indeterminate state
3. Check the styling of the indeterminate state.
4. Check out the branch
5. Run `npm run test` in `packages/core`
6. Make sure all tests pass.



[CDEP-1589]: https://tegel.atlassian.net/browse/CDEP-1589?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
